### ISSUE TITLE
Follow Button working on Local but not on Vercel

### ISSUE THREAD

**Issue by lboone** [codinginflow/nextjs-15-social-media-app#5]
Hi, when i press the follow button on the Vercel version it does not work, however it works perfectly fin on the local version.
I get the following error in Vercel logs:

PrismaClientValidationError: 
Invalid `prisma.follow.upsert()` invocation:

{
  where: {
    followerId_followingId: {
      followerId: "pvc3cteknhhexksn",
+     followingId: String
    }
  },
  create: {
    followerId: "pvc3cteknhhexksn",
    followingId: undefined
  },
  update: {}
}

Argument `followingId` is missing.
    at An (/var/task/node_modules/@prisma/client/runtime/library.js:114:7526)
    at _n.handleRequestError (/var/task/node_modules/@prisma/client/runtime/library.js:121:7392)
    at _n.handleAndLogRequestError (/var/task/node_modules/@prisma/client/runtime/library.js:121:7057)
    at _n.request (/var/task/node_modules/@prisma/client/runtime/library.js:121:6741)
    at async l (/var/task/node_modules/@prisma/client/runtime/library.js:130:9355) {
  clientVersion: '5.17.0'
}

**Comment by robertdluigi** [comment]
Is your code looking like this?

```
export async function POST(
  req: Request,
  { params: { userId } }: { params: { userId: string } },
) {
  try {
    const { user: loggedInUser } = await validateRequest();

    if (!loggedInUser) {
      return Response.json({ error: "Unauthorized" }, { status: 401 });
    }

    await prisma.$transaction([
      prisma.follow.upsert({
        where: {
          followerId_followingId: {
            followerId: loggedInUser.id,
            followingId: userId,
          },
        },
        create: {
          followerId: loggedInUser.id,
          followingId: userId,
        },
        update: {},
      }),
      prisma.notification.create({
        data: {
          issuerId: loggedInUser.id,
          recipientId: userId,
          type: "FOLLOW",
        },
      }),
    ]);

    return new Response();
  } catch (error) {
    console.error(error);
    return Response.json({ error: "Internal server error" }, { status: 500 });
  }
}
```
For me it works fine both locally and deployed on Vercel

**Comment by lboone** [comment]
I figured out what the problem was.  When i created my api routes i used a capital letter for the userid.  So I created it like this:

api/users/[userId]/ etc...

But when I deployed the site, for some reason the folder structure on the deployed site turned the userId into userid.

When the component extracted the userid, it was returning a lowercase i but my code was looking for an uppercase i.

Strange, but once I figured it out, i changed all the code to look for a lowercase i  and it worked.  thanks for your response.
